homeassistant/components/roborock/coordinator.py

```diff
@@ -125,12 +130,21 @@ def _set_current_map(self) -> None:
 
     async def get_maps(self) -> None:
         """Add a map to the coordinators mapping."""
+        self.maps = await self.roborock_storage.async_load_map_info(self.duid)
         maps = await self.api.get_multi_maps_list()
+        update = False
         if maps and maps.map_info:
             for roborock_map in maps.map_info:
-                self.maps[roborock_map.mapFlag] = RoborockMapInfo(
-                    flag=roborock_map.mapFlag, name=roborock_map.name, rooms={}
-                )
+                # To prevent weirdness - if there is a map without a name
+                # - set its name to its flag
+                if roborock_map.mapFlag not in self.maps:
+                    map_name = roborock_map.name or str(roborock_map.mapFlag)
+                    self.maps[roborock_map.mapFlag] = RoborockMapInfo(
+                        flag=roborock_map.mapFlag, name=map_name, rooms={}
+                    )
+                    update = True
+        if update:
```

Comment:
What is expected behavior if the API no longer returns a map? I might expect it to be removed from the local disk cache?

Reply:
Maybe? It probably should be - and the entity should be removed as well. Is the proper way to remove the entity just using the entity registry?